### ISSUE TITLE
Order "Report type" options

### ISSUE THREAD

**Issue by noahmanger** [18F/openFEC-web-app#360]
And maybe think of a better way to group these or something. Not sure at the moment.

![screen shot 2015-07-23 at 10 22 52 am](https://cloud.githubusercontent.com/assets/1696495/8857193/dd135faa-3124-11e5-8e46-70421166f1da.png)

**Comment by noahmanger** [comment]
Update: I see that they're actually grouped in a logical way (all monthly reports are together, for instance). But maybe those should be broken out into their own dropdown?

**Comment by noahmanger** [comment]
Is this still something we want to do? If so @PaulClark2 could you share the order you shared earlier? I can't find it.

**Comment by PaulClark2** [comment]
@noahmanger  will also send to you in Slack DM

**Form Types Selections**:  
Statements Of Organization - Form 1
Multicandidate Status - Form 1M
Statements Of Candidacy - Form 2
Congressional Candidate Financial Reports - Form 3
Presidential Financial Reports - Form 3P
PAC and Party Financial Reports - Form 3X
Bundled Contributions Reports - Form 3L
Convention Financial Reports - Form 4
Independent Expenditure Reports and Notices - Forms 5 & 24
Contributions and Loans Notices - Form 6
Communication Cost Reports - Form 7 
Debt Settlement Plans - Form 8
Electioneering Communications Notices - Form 9
Inaugural Committee Donation Reports - Form 13
Miscellaneous Submissions
Requests for Additional Information
Expenditure of Personal Funds Notices - Form 10
Opposition Personal Funds Notices - Form 11
Suspension of Increased Limits Notices - Form 12

**Report Type Selections** (including or excluding the report code)
Monthly Reports
"M2" February Monthly
"M3" March Monthly
"M4" April Monthly
"M5" May Monthly
"M6" June Monthly
"M7" July Monthly
"M8" August Monthly
"M9" September Monthly
"M10" October Monthly
"M11" November Monthly
"M12" December Monthly
"YE" Year-End

Quarterly Reports
"Q1" April Quarterly
"Q2" July Quarterly
"Q3" October Quarterly
"YE" Year-End

Semi-annual Reports 
"MY" Mid-Year
"YE" Year-End

Election Sensitive Reports
"12P" Pre-Primary
"12C" Pre-Convention
"12G" Pre-General
"12R" Pre-Runoff
"12S" Pre-Special
"30G" Post-General
"30R" Post-Run-Off
"30S" Post-Special
"30P" Post-Primary
"60D" Post-Election

Other Reports 
"TER" Termination
"24" 24-Hour Notification
"48" 48-Hour Notification
"90D" Post-Inaugural
"90S" Post-Inaugural Supplement
"CA" Comprehensive Amendment
"ADJ" Comprehensive Adjusted Amendment

Bundling Reports
"M7S" July Monthly / Semi-Annual
"MSA" Monthly Semi-Annual (MY)
"MSY" Monthly Semi-Annual (YE)
"MYS" Monthly Year-End / Semi-Annual 
"Q2S" July Quarterly/Semi-Annual
"QSA" Quarterly Semi-Annual (MY)
"QYE" Quarterly Semi-Annual (YE)
"QYS" Quarterly Year-End / Semi-Annual
"QMS" Quarterly Mid-Year / Semi-Annual

**Comment by noahmanger** [comment]
There's a few I see in our list that aren't included here: 
```    
('10D', 'PRE-ELECTION'),
('10G', 'PRE-GENERAL'),
('10P', 'PRE-PRIMARY'),
('10R', 'PRE-RUN-OFF'),
('10S', 'PRE-SPECIAL'),
('30D', 'POST-ELECTION'),
```
Where do those go?

**Comment by PaulClark2** [comment]
@noahmanger  These are ancient report codes but all of them go in the "Election Sensitive Reports" section.

**Comment by noahmanger** [comment]
Should we designate that those are old codes? Their names are the same as the others in the "Election Sensitive Reports" so curious how to deduplicate.

**Comment by noahmanger** [comment]
Also, do we want to break these report types out into their own filters? It would make for several more filters, but each one would be far more usable.

**Comment by noahmanger** [comment]
Also, with Forms 5 and 24 being "Independent Expenditure Reports and Notices" -- do we just want one option that fills in two form types? Or include each one as a separate option?

**Comment by noahmanger** [comment]
More questions:
- [ ] Is there no "M1 - January Monthly"?
- [ ] Are the codes for Monthly Year-End and Quarterly Year-End and Semi-Annual Year-End all "YE"? Should we just list it once?

**Comment by PaulClark2** [comment]
The descriptions for these 4 are duplicated  by newer report type codes.   
('10G', 'PRE-GENERAL'),
('10P', 'PRE-PRIMARY'),
('10R', 'PRE-RUN-OFF'),
('10S', 'PRE-SPECIAL'),

I think we have two options and I don't have a preference for option 1 or 2.  I think either would work.  
1) return both report type codes when someone select one of these 4 report descriptions 
Pre-General = 10G and 12G
Pre-Primary = 10P and 12P
Pre-Runoff  = 10R and 12R; or 

2) mark these 4 with a date range
10G	1982 (last report year used)
10P	1982 (last report year used)
10R	1980 (last report year used)
10S	1980 (last report year used)

 

The other two 10D and 30D are not duplicated by newer reports.  
('10D', 'PRE-ELECTION'),
('30D', 'POST-ELECTION'),

**Comment by PaulClark2** [comment]
Forms 5 and 24 being "Independent Expenditure Reports and Notices":   This might we a usability question.  We group them like this because we assumed users would want to see all 24-hour and 48-hour reports of independent expenditures and would be less concerned if if the filer was a registered committee (F24) or a person/group (F5).   It makes sense to me to keep them together but I'm willing to listen to other ideas

**Comment by PaulClark2** [comment]
There is no M1 report.   Oddity of our monthly report naming convention is the report covers financial activity of the prior month.  So the M2 - February Monthly discloses financial activity for the month of January.   

The Year-End code, YE, for monthly, quarterly and semi-annual reports is the same.   We use it three times because we have monthly, quarterly and semi-annual report types in 3 groups.  So that it's clear each type of filer files a year-end report.   I'm open to modifying that.  However, if we make separate filters for each group then we'd need YE to be in each filter.  

If we have multiple filers, do you think users would be confused?  Good filter labels would help.   I agree having the report types in short lists probably helps people.  

Will they be able to select YE and M2?   I ask this because most presidential filers file quarterly in the odd year before the election but file monthly during the election year.  So if you look at the current presidential candidates they are filing quarterly reports but in February 2016 they will start filing monthly reports.

**Comment by noahmanger** [comment]
Cool. This all makes sense. Based on how the code is now, it will be easiest if we restrict each option to just one report code each (@jmcarp feel free to jump in and say it's super easy to do otherwise), but I think that's fine. 

Here's my suggestion for how to label the fields in question:

- Pre-Election 
- Post-Election
- Pre-General (last used: 1982)
- Pre-Primary (last used: 1982)
- Pre-Runoff (last used: 1980)
- Pre-Special (last used: 1980)

- Independent Expenditure Reports and Notices (by a registered committee)
- Independent Expenditure Reports and Notices (by a person or group)

And then I the hearing how YE is the same for all, I think it makes sense to break each type out into a separate filter (monthly, quarterly, etc.) and include Year-End in each. 

The filters would all be additive, so you could do YE and M2 and it would show all Year End reports and all February Monthly reports.

**Comment by PaulClark2** [comment]
I like it!

**Comment by noahmanger** [comment]
PR updated to reflect these changes https://github.com/18F/openFEC-web-app/pull/604

**Comment by jmcarp** [comment]
Merged #604.
